src/Functions/FunctionsHashing.h

```diff
@@ -1193,8 +1199,9 @@ class FunctionAnyHash : public IFunction
     {
         for (size_t i = 0, size = column->size(); i < size; ++i)
         {
-            StringRef bytes = column->getDataAt(i);
-            const ToType h = apply(key, bytes.data, bytes.size);
+            const ToType h = column->isNullAt(i) ? static_cast<ToType>(impl::EMPTY_HASH) :
+                apply(key, column->getDataAt(i).data, column->getDataAt(i).size);
```

Comment:
@aiven-sal That does not matter.

Reply:
@aiven-sal check this https://github.com/ClickHouse/ClickHouse/blob/3f0c3602bf0b39d72d85a6fb875ba093763222a1/src/Functions/FunctionsHashing.h#L1436

Reply:
Something like this is already a problem without this PR:
```
SELECT cityHash64([1]) AS h
2963573356302499406

SELECT cityHash64([toNullable(1)]) AS h
6974130533670503360
```
The first dispatches to executeIntType, second - executeGeneric, hence different results. Without the array, results are the same - inconsistent.

A simple thing we can do with this is: add executeNullable() (called from executeAny()), which would first call executeAny() for the nested column, then iterate over the null map and overwrite hashes corresponding to nulls with constant.

This will make hash of `toNullable(x)` always be the same as has of `x`, no matter how many levels of arrays/tuples/maps it's wrapped in.